projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -1404,3 +1412,37 @@ public static function register_block_metadata_collection() {
 	 */
 	add_filter( 'jetpack_block_editor_enable_upgrade_nudge', '__return_true' );
 }
+
+/**
+ * Overrides the `wp-block-editor` script with a hotfix to prevent a performance issue.
+ *
+ * @param WP_Scripts $scripts WP_Scripts instance.
+ */
+function jetpack_hotfix_block_editor_script( $scripts ) {
+	if ( ! defined( 'GUTENBERG_VERSION' ) ) {
+		return;
+	}
+
+	if ( ! in_array( GUTENBERG_VERSION, array( '19.9.0', '20.0.0', '20.1.0' ), true ) ) {
+		return;
+	}
+
+	$path         = gutenberg_dir_path() . 'build/block-editor/index.min.js';
+	$handle       = 'wp-block-editor';
+	$asset_file   = substr( $path, 0, - ( strlen( '.js' ) ) ) . '.asset.php';
+	$asset        = file_exists( $asset_file ) ? require $asset_file : null;
+	$dependencies = isset( $asset['dependencies'] ) ? $asset['dependencies'] : array();
+	$version      = isset( $asset['version'] ) ? $asset['version'] : GUTENBERG_VERSION;
+	gutenberg_override_script(
+		$scripts,
+		$handle,
+		Assets::get_file_url_for_environment(
+			'_inc/block-editor-hotfix-' . GUTENBERG_VERSION . '.min.js',
+			'_inc/block-editor-hotfix-' . GUTENBERG_VERSION . '.min.js'
```

Comment:
Given that we only have minified files, would it be possible to simplify things and remove usage of the `get_file_url_for_environment` method?

Reply:
Done in https://github.com/Automattic/jetpack/pull/41309/commits/b5d48b1ef5030a124fc7d04af7d79abac97e33cb